### PR TITLE
Show custom icons in Component Browser

### PR DESCRIPTION
**Pull request by akavel**
### Pull Request Description

Show custom icons in Component Browser for entries that have a non-empty `Icon` section in their docs with the section's body containing a name of a predefined icon.

https://www.pivotaltracker.com/story/show/182584336

#### Visuals

A screenshot of a couple custom icons in the Component Browser:

Screenshot 2022-07-27 at 15 55 33

### Important Notes

 - The PR assigns icon names to four items in the standard library, but only three of them are shown in the Component Browser because of [a parsing bug in the Engine](https://www.pivotaltracker.com/story/show/182781673).
 - Icon names are assigned only to four items in the standard library because only two currently predefined icons match entries in the currently defined Virtual Component Groups. Adjusting the definitions of icons and Virtual Component Groups is covered by [a different task](https://www.pivotaltracker.com/story/show/182584311).
 - A bug in the documentation of the Enso protocol message `DocSection` is fixed. A `text` field in the `Tag` interface is renamed to `body` (this is the field name used in Engine).

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide dist` and `./run ide watch`.

[ci no changelog needed]

### CODE REVIEW

File: distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso [hunk 1]
```diff
@@ -140,6 +140,8 @@ type Table
          Select columns with the same names as the ones provided.
 
              table.select_columns (By_Column [column1, column2])
+
+       Icon: select-column
```
Comment (radeusgd) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso]: If this is modified, then I think the analogous one in `Standard/Table` should be modified too, as these functions should be kept consistent, including the documentation.

File: lib/rust/prelude/src/serde.rs [hunk 2]
```diff
@@ -19,6 +19,18 @@ where
...
+#[cfg(feature = "serde_json")]
+pub fn deserialize_null_as_default<'d, Ret, D>(d: D) -> Result<Ret, D::Error>
+where
+    for<'e> Ret: Default + Deserialize<'e>,
+    D: serde::Deserializer<'d>, {
+    let option_value = Option::deserialize(d)?;
+    Ok(option_value.unwrap_or_default())
+}
+
+
```
Comment (farmaazon) [inline lib/rust/prelude/src/serde.rs]: If the field is missing, then I assume this behaves as it would be null?

**Review by farmaazon — APPROVED** [review]

File: lib/rust/prelude/src/serde.rs (hunk 2 above)
Comment (akavel) [reply lib/rust/prelude/src/serde.rs]: If and only if used together with `#[serde(default)]`. I moved the test of this function into a runnable documentation test to make it more clear how it is expected to be used.

**Review by radeusgd — APPROVED** [review]
The stdlib changes look good to me. I assume we will be adding icons to other methods later on.

**Review by 4e6 — APPROVED** [review]

**Review by wdanilo — APPROVED** [review]

**Comment by akavel** [comment]
> I assume we will be adding icons to other methods later on.

@radeusgd Yes - see: https://www.pivotaltracker.com/story/show/182584311. Please note that we talked about this ticket with @jdunkerley with an understanding that this will be planned by him / the Library Team. That said, this may need some clarification / sync between teams, possibly during the next week's meetup. The CB Epic Team had a quick rehash of this with @xvcgreg today, I assume he's going to stay on top of this; but it might be good if you and me also try to bring it up during the meetup. With that said, CB Team's current intuition is it might be best to address this ticket after September, once the CB starts getting used and the contents of the Favorites start getting tested & fleshed out.

<blockquote><div><strong><a href="https://www.pivotaltracker.com/story/show/182584311">Add icon sections in documentation of chosen standard library entries. - Pivotal Tracker</a></strong></div></blockquote>

**Pull request merged**